crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -1615,8 +1748,28 @@ impl<'db> BindingError<'db> {
             }
 
             Self::MissingArguments { parameters } => {
+                let s = if parameters.0.len() == 1 { "" } else { "s" };
+
+                if let Some(union_diag) = diag {
+                    let sub = union_diag.sub(
+                        context.db(),
+                        &MISSING_ARGUMENT,
+                        node,
+                        format_args!(
+                            "no argument{s} provided for \
```

Comment:
I think this message doesn't quite work grammatically when prepended with "... because ..." -- needs to be "no argument{s} _was_ provided for ..."